packages/protocol/contracts/governance/DoubleSigningSlasher.sol

```diff
@@ -0,0 +1,163 @@
+pragma solidity ^0.5.3;
+
+import "openzeppelin-solidity/contracts/ownership/Ownable.sol";
+import "openzeppelin-solidity/contracts/math/SafeMath.sol";
+
+import "../common/Initializable.sol";
+import "../common/UsingRegistry.sol";
+import "../common/UsingPrecompiles.sol";
+
+contract DoubleSigningSlasher is Ownable, Initializable, UsingRegistry, UsingPrecompiles {
+  struct SlashingIncentives {
+    // Value of LockedGold to slash from the account.
+    uint256 penalty;
+    // Value of LockedGold to send to the observer.
+    uint256 reward;
+  }
+
+  SlashingIncentives public slashingIncentives;
+  mapping(bytes32 => bool) isSlashed;
+
+  event SlashingIncentivesSet(uint256 penalty, uint256 reward);
+
+  function initialize(address registryAddress, uint256 _penalty, uint256 _reward)
+    external
+    initializer
+  {
+    _transferOwnership(msg.sender);
+    setRegistry(registryAddress);
+    _setSlashingIncentives(_penalty, _reward);
+  }
+
+  // Require that `reward` is less than `penalty`.
+  function setSlashingIncentives(uint256 penalty, uint256 reward) external onlyOwner {
+    _setSlashingIncentives(penalty, reward);
+  }
+
+  // Require that `reward` is less than `penalty`.
+  function _setSlashingIncentives(uint256 penalty, uint256 reward) internal {
+    require(penalty > reward, "Penalty has to be larger than reward");
+    slashingIncentives.penalty = penalty;
+    slashingIncentives.reward = reward;
+    emit SlashingIncentivesSet(penalty, reward);
+  }
+
+  // Given two RLP encoded blocks, calls into a precompile that requires that
+  // the two block hashes are different, have the same height, have a
+  // quorum of signatures, and that `signer` was part of the quorum.
+  // Returns the block number of the provided blocks.
+  function eval(
```

Comment:
I think ideally the `eval` functions would expose `external view`s so observers can more effectively automate their claims